articles/introduction-to-scikit-learn-in-python/index.md

```diff
@@ -0,0 +1,297 @@
+
+### Table of Contents
+
+1. [Introduction](#introduction)
+2. [Installing the Scikit-Learn Library](#installing-the-scikit-learn-library)
+3. [Dataset Transformations using sklearn](#dataset-transformations-using-sklearn)
+4. [Scikit-Learn for Standardization](#scikit-learn-for-standardization)
+5. [Scikit-Learn for Normalization](#scikit-learn-for-normalization)
+6. [Scikit-Learn for Encoding Categorical Features](#scikit-learn-for-encoding-categorical-features)
+7. [Scikit-Learn for Filling Missing Values](#scikit-learn-for-filling-missing-values)
+8. [Conclusion](#conclusion)
+9. [Further Readings](#further-readings)
+
+
+### Introduction
+
+SciKit-Learn (often referred to as sklearn) provides a wide array of statistical models and machine learning. sklearn, unlike most modules, is written in Python and not in C. Although written in Python, sklearn's performance is attributed to its usage of NumPy for high-performance linear algebra and array operations. SciKit-Learn was written as a part of Google's Summer of Code project and has since made lives easier for thousands of Python centered data scientists worldwide. This part of the series focuses on introducing the library and focusing on one aspect - dataset transformations, an important and crucial step to go through before building a prediction model.
+
+### Installing the Scikit-Learn Library
+
+Scikit-Learn requires the following libraries to be pre-installed: NumPy, SciPy, Matplotlib, IPython, Sympy, and Pandas. Let's go ahead and install them from the terminal using pip (works only for Windows)
+
+\>> pip install numpy
+\>> pip install scipy
+\>> pip install matplotlib
+\>> pip install ipython
+\>> pip install sympy
+\>> pip install pandas
+
+Now that we've installed the dependent libraries let us install Scikit-Learn.
+
+\>> pip install scikit-learn
+
+Let's check if Scikit-Learn can be accessed using Python.
+
+```py
+import sklearn
+```
+Yes, it works!
+
+### Dataset Transformations using sklearn
+
+An essential component for building a machine learning algorithm is data. A lot of work goes into preparing the data so that it can be fed to the model. This is called data preprocessing. Data preprocessing tasks can range from a mere change in notation to changing a continuous variable to a categorical variable. The sklearn.preprocessing package provides various functions and classes to change the representation of certain variables to be better suited for the estimators down the model pipeline. So, let's go ahead and look at the methods that Scikit-Learn offers, which help in data preprocessing and transformation. Before that, let's import the sklearn.preprocessing package.
+
+```py
+from sklearn import preprocessing
+```
+
+### Scikit-Learn for Standardization
+
+Standardization is an essential task for [distance based models](https://ir.library.oregonstate.edu/concern/graduate_thesis_or_dissertations/zw12z7835) so that one particular feature does not dominate over the other. 
```

Comment:
What are distance-based models?

Reply:
I've added a resource to this. The link redirects to a conference paper whose abstract explains what distance-based models are in detail. Would the attached resource be enough or should I add more?

Reply:
Just explain them in a line or two. It is important to understand the context presented here.

Reply:
I've made the changes needed. Thanks!